x-pack/plugins/security_solution/public/resolver/store/middleware/index.ts

```diff
@@ -33,7 +33,7 @@ export const resolverMiddlewareFactory: MiddlewareFactory = (dataAccessLayer: Da
       next(action);
 
       resolverTreeFetcher();
-      relatedEventsFetcher();
+      relatedEventsFetcher(action);
```

Comment:
can you remove the `action` parameter and check for changes to `isLoadingMoreNodeEventsInCategory` and `isLoadingNodeEventsInCategory` instead? That way the reducer controls the logic of deciding when to request data. The middleware only has to respond to a change in state.